src/main/java/org/mule/extension/introspection/declaration/fluent/BaseDeclaration.java

```diff
@@ -70,6 +73,59 @@ public T addCapability(Object capability)
     }
 
     /**
+     * Returns a map with the currently set model properties. Notice
+     * that this map is mutable and not thread-safe. This method
+     * is to be used when you need to access all the properties. For individual
+     * access use {@link #addModelProperty(String, Object)} or
+     * {@link #getModelProperty(String)} instead
+     *
+     * @return a {@link Map} with the current model properties. Might be empty but will never by {@code null}
+     */
+    public Map<String, Object> getModelProperties()
+    {
+        return modelProperties;
+    }
+
+    /**
+     * Returns the model property registered under {@code key}
+     *
+     * @param key the property's key
+     * @param <T> the generic type for the response value
+     * @return the associated value or {@code null} if no such property was registered
+     */
+    public <T> T getModelProperty(String key)
+    {
+        return (T) getModelProperties().get(key);
+    }
+
+    /**
+     * Associates the {@code key} with the {@code value} property
+     *
+     * @param key   the property's key
+     * @param value the property value
+     * @throws IllegalArgumentException if key is {@code null}, empty or already registered. Also if {@code value} is {@code null}
+     */
+    public void addModelProperty(String key, Object value)
```

Comment:
Is needed to throw an exception when the key was already registered? (maybe a post processor wants to override a value)